File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/ui/MainActivityModel.java
package com.byagowi.persiancalendar.ui;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainActivityModel extends ViewModel {
    public final MutableLiveData<Void> preferenceUpdateHandler = new MutableLiveData<>();

    void preferenceIsUpdated() {
        preferenceUpdateHandler.postValue(null);
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/ui/MainActivityModel.kt
package com.byagowi.persiancalendar.ui

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainActivityModel : ViewModel() {
    val preferenceUpdateHandler = MutableLiveData<Void>()

    internal fun preferenceIsUpdated() {
        preferenceUpdateHandler.postValue(null)
    }
}
